### ISSUE TITLE
False positive on `return` for remove_parentheses

### ISSUE THREAD

**Issue by A4-Tacks** [rust-lang/rust-analyzer#21063]
**rust-analyzer version**: rust-analyzer 0.4.2671-standalone

**rustc version**: rustc 1.93.0-nightly (f15a7f385 2025-11-04)

**code snippet to reproduce**:
```rust
fn main() {
    let _x = true && !$0(return) || true;
}
```

**Current**:

Assist applicable `Remove redundant parentheses`

```rust
fn main() {
    let _x = true && !return || true;
}
```

This returns a closure!

**Expect**

Assist not applicable

This seems a bit complicated because `cond || (return);` -> `cond || return;` It's expected

**Comment by A4-Tacks** [comment]
Since https://github.com/rust-lang/rust-analyzer/pull/21092#issuecomment-3565872347 has not been fixed, reopened this

**Comment by asuto15** [comment]
@A4-Tacks 
I've made a fix(#21090) for the test cases that discussed in #21092.

If you have additional edge cases or expected behaviors, please let me know. I’m happy to refine it further.
